opencog/truthvalue/AttentionValue.h

```diff
@@ -35,8 +35,10 @@ namespace opencog
  */
 
 //! stores attention in three components: short-term, long-term and very long-term
+//! This class is currently read-only, and assumed to be thread-safe. Beware of
+//! possible race conditions if it is changed
 class AttentionValue;
-typedef std::shared_ptr<AttentionValue> AttentionValuePtr;
+typedef std::shared_ptr<const AttentionValue> AttentionValuePtr;
```

Comment:
Linas, what Hedayat is working on is parallel execution of Agents and other background processes. The goal is to run ECAN and background PLN in parallel, as well as processing incoming requests such as ROS messages, chat, etc. Right now Misgana and Matt Iklé are tuning ECAN at the Atom level, and AFAIK they're not changing the immutable nature of the AV object. I don't know of any plans to change that in the near future, but perhaps you or @bgoertzel do?

If we're not changing that object, why not mark it as read-only?